crates/holochain/tests/tests/publish/mod.rs

```diff
@@ -29,56 +31,62 @@ async fn publish_terminates_after_receiving_required_validation_receipts() {
 
     let ((alice,), (bobbo,), (carol,), (danny,), (emma,), (fred,)) = apps.into_tuples();
 
+    let apps = [alice, bobbo, carol, danny, emma, fred];
+
+    for c in conductors.iter() {
+        c.holochain_p2p()
+            .test_set_full_arcs(apps[0].dna_hash().to_k2_space())
+            .await;
+    }
+
     let action_hash: ActionHash = conductors[0]
-        .call(&alice.zome(TestWasm::Create), "create_entry", ())
+        .call(&apps[0].zome(TestWasm::Create), "create_entry", ())
         .await;
 
-    // Wait until they all see the created entry, at that point validation receipts should be getting sent soon
-    await_consistency(60, [&alice, &bobbo, &carol, &danny, &emma, &fred])
-        .await
-        .unwrap();
+    tokio::time::timeout(
+        std::time::Duration::from_secs(60),
+        async {
+            loop {
+                // make sure all our workflows keep running periodically
```

Comment:
Not following this logic. Which workflow needs to re-run? Is this because the peers might not have all discovered each other yet and so validation receipts might not get sent on the first try?

I wonder if the test could assert that peer discovery is complete for all nodes before creating the entry that we want receipts for? That way we know that all validators should be able to make a connection to us.

Reply:
I have to say I don't really care. This seems to make the test not flaky in the limited number of CI runs I've done. Is it really worth the time to dig further?

Tell me it is, and I'll happily dig further : )

Reply:
It's worth it to me, that these tests are simple and reflect as closely as possible how Holochain will behave in the real world. Continuously publishing to try to get validation receipts for previous actions would not be something I'd want to have to ask a user to do if something wasn't working.

I believe that it's made it not flaky because I can see the logic but I'd prefer:
- Guarantee bootstrapping and arcs ready to go before creating the single action. We're having to do this in quite a few tests, maybe we just need that functionality in sweettest.
- If we can't make that work, get the cell from each conductor and use its workflow triggers to force running the validation receipt workflows again.

Reply:
Done: https://github.com/holochain/holochain/pull/4834/commits/df0d83c8e857b80a4ed09d4fc1625734293e722d